test/extended/builds/cluster_config.go

```diff
@@ -0,0 +1,111 @@
+)
+
+// e2e tests of the build controller configuration.
+// These are tagged [Serial] because each test modifies the cluster-wide build controller config.
+var _ = g.Describe("[Feature:Builds][Serial][Slow] alter builds via cluster configuration", func() {
+	defer g.GinkgoRecover()
+	var (
+		buildFixture           = exutil.FixturePath("testdata", "builds", "test-build.yaml")
+		defaultConfigFixture   = exutil.FixturePath("testdata", "builds", "cluster-config.yaml")
+		blacklistConfigFixture = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-blacklist.yaml")
+		searchConfigFixture    = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-search.yaml")
+		whitelistConfigFixture = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-whitelist.yaml")
+		oc                     = exutil.NewCLI("build-cluster-config", exutil.KubeConfigPath())
+	)
+
+	g.Context("", func() {
+
+		g.BeforeEach(func() {
+			exutil.PreTestDump()
+		})
+
+		g.JustBeforeEach(func() {
+			g.By("waiting for default service account")
+			err := exutil.WaitForServiceAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()), "default")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("waiting for builder service account")
+			err = exutil.WaitForServiceAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()), "builder")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			oc.Run("create").Args("-f", buildFixture).Execute()
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+				exutil.DumpConfigMapStates(oc)
+			}
+			oc.AsAdmin().Run("apply").Args("-f", defaultConfigFixture).Execute()
+		})
+
+		g.Context("registries config context", func() {
+
+			g.It("should default registry search to docker.io for image pulls", func() {
+				g.By("apply default cluster configuration")
+				err := oc.AsAdmin().Run("apply").Args("-f", defaultConfigFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("waiting 1s for build controller configuration to propagate")
+				time.Sleep(1 * time.Second)
+				g.By("starting build sample-build and waiting for success")
+				// Image used by sample-build (centos/ruby-22-centos7) is only available on docker.io
+				br, err := exutil.StartBuildAndWait(oc, "sample-build")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertSuccess()
+			})
+
+			g.It("can override registry search for image pulls", func() {
+				g.By("apply image search cluster configuration")
+				err := oc.AsAdmin().Run("apply").Args("-f", searchConfigFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("waiting 1s for build controller configuration to propagate")
+				time.Sleep(1 * time.Second)
+				g.By("starting build sample-build and waiting for failure")
+				br, err := exutil.StartBuildAndWait(oc, "sample-build")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertFailure()
```

Comment:
failures can happen for a lot of reasons...you should still ensure the search was resolved in the way you expected.

Reply:
Added check to ensure log has message "no such image"